Content.Client/DistortionMap/DistortionMap.cs

```diff
@@ -0,0 +1,116 @@
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Collections.Generic;
+using Robust.Client.Graphics;
+using Robust.Shared.Enums;
+using Robust.Shared.IoC;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Maths;
+
+namespace Content.Client.DistortionMap;
+
+class DistortionMapOverlay : Overlay
+{
+    [Dependency] private readonly IClyde _clyde = default!;
+    [Dependency] private readonly IPrototypeManager _prototypeManager = default!;
+
+    public override OverlaySpace Space => OverlaySpace.WorldSpace;
+    public override bool RequestScreenTexture => true;
+
+    private Dictionary<IClydeViewport, DistMapInstance> _maps = new();
+    private ShaderInstance? _shd = default;
+
+    public DistortionMapOverlay()
+    {
+        IoCManager.InjectDependencies(this);
+
+        ZIndex = 99999;
+
+        _shd = _prototypeManager
+            .Index<ShaderPrototype>("Distortion")
+            .Instance()
+            .Duplicate();
+
+        if (_shd is null)
+            throw new Exception("unable to create distortion map shader");
+    }
+
+    public const float MapSizeDivisor = 2.0f;
+
+    public DistMapInstance? CreateMap(IClydeViewport vp, float szDiv = MapSizeDivisor)
```

Comment:
- This PR **has not been successfully tested** with `szDiv` set to anything other than `1.0f`.
- `szDiv` should be configurable via some sort of setting as a performance knob.